mbed-client-pal/Source/Port/Reference-Impl/OS_Specific/ZephyrOS/RTOS/pal_plat_rtos.c

```diff
@@ -286,27 +273,17 @@ palStatus_t pal_plat_osTimerStart(palTimerID_t timerID, uint32_t millisec)
         return PAL_ERR_RTOS_VALUE;
     }
 
-    uint64_t uptime = k_uptime_get();
+    int64_t uptime = k_uptime_get();
 
     k_mutex_lock(&timer_data->lock, K_FOREVER);
 
-    timer_data->timeout = (uptime + millisec);
+    timer_data->timeout = (uptime + (int64_t)millisec);
```

Comment:
adding signed integers is very dangerous in C as overflow here is an undefined behavior